tests/test_base_device.py

```diff
@@ -729,6 +729,66 @@ def test_num_wires_property(self):
         assert dev.num_wires == len(wires)
 
 
+class TestTrackerFunctionality:
+    def test_tracker_batched(self):
+        """Test that the tracker works for batched circuits"""
+        dev = qml.device("default.qubit", wires=1, shots=10000)
+        qiskit_dev = QiskitDevice2(wires=1, backend=AerSimulator(), shots=10000)
+
+        x = np.array(0.1, requires_grad=True)
+
+        @qml.qnode(dev, diff_method="parameter-shift")
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.Z(0))
```

Comment:
This is a single very simple example of what we might track. It would be nice to include some tests that have more complex returns. I would also question the assumption that comparing the tracker on the Qiskit device to the tracker on `default.qubit` for the same circuit is a good test in this case, because, since the devices behave differently, we shouldn't expect their trackers to match consistently.

Here are some examples of possible return statements where I think things are off (the circuits will need to just be called instead of calling `grad` on them, since they have multiple outputs):

- `return qml.expval(qml.Z(0)), qml.expval(qml.X(1)), qml.counts(qml.X(2))` : for `default.qubit`, this should (and does) assume 1 execution, because everything commutes. But `expval` and `counts` go to different tapes, so I would expect the Qiskit device to show 2 executions, since ultimately two tapes are executed when calling `qiskit_circuit(x)`. In this case, the two devices do match, but I would expect them not to.
- `return qml.expval(qml.Z(0)), qml.expval(qml.X(0))`: here I would expect the results for this device and `default.qubit` to match, but instead it splits the `qiskit_device` return into multiple calls. I think this is less of a problem, because the `tracker.history` still has reasonable results, it's just that they are split up differently.

Reply:
>`return qml.expval(qml.Z(0)), qml.expval(qml.X(1)), qml.counts(qml.X(2))` : for default.qubit, this should (and does) assume 1 execution, because everything commutes. But expval and counts go to different tapes, so I would expect the Qiskit device to show 2 executions, since ultimately two tapes are executed when calling qiskit_circuit(x). In this case, the two devices do match, but I would expect them not to.

I am a little confused. When I ran this, I got `'executions': 2` for the qiskit device and `'executions:1'` for the default device (Which is... what you want right? This is the intended behaviour on my part as well). I can add an additional test case for this (in fact... I should... my bad... let me do that...) but maybe this is bugged somehow?

>`return qml.expval(qml.Z(0)), qml.expval(qml.X(0))`: here I would expect the results for this device and default.qubit to match, but instead it splits the qiskit_device return into multiple calls. I think this is less of a problem, because the tracker.history still has reasonable results, it's just that they are split up differently.

Here I expect the `qiskit_device` to be split up as well, since it's going to make two separate calls to the Estimator. 

In general, my intention was to have the `executions` to match as closely as possible to the number of calls to the IBM backends, so users don't have to check their IBM account dashboards to know how many calls occurred (although, they really should for peace of mind). So far I think this is pretty much working as intended (although the tests cases do not really "test" for that... which is... very fixable!)

Reply:
Discussed offline, concluded that the tracker treats `latest` differently than I was expecting, but this is alright given how the device differs from the PennyLane devices. Specifically, `latest` reflects the latest call to run on Qiskit, not the latest call of the QNode.

Is summary, the tracker is counting what we want it to count, but the bookkeeping structure is different than what I expected. This seems like more work than it's worth to look into, and the existing version is accurate.